src/DB/Table/ShippingRateTable.php

```diff
@@ -26,10 +26,13 @@ protected function get_install_query(): string {
 CREATE TABLE `{$this->get_sql_safe_name()}` (
     id bigint(20) NOT NULL AUTO_INCREMENT,
     country varchar(2) NOT NULL,
+    method varchar(30) NOT NULL DEFAULT 'flat_rate',
```

Comment:
> Would it make more sense to do the same and then set it to a default of flat_rate in the migrate class?

I initially intended to do that but this sounded like a much simpler and fool-proof way so I went with a default. I do agree that updating it manually would make the table definition look better.

But even if we set the default as `''` it would mean that the table definition is wrong because there should never be an empty value set in the `method` field. In this case, the actual default value that we expect is `flat_rate`.

We can skip dropping the `DEFAULT` in the migration class as you suggested but still keep the default as `flat_rate`. Although I'm not sure if that would be the right call because we don't expect a `NOT NULL` column to have a default.

I guess either way we're going to have some minor inconsistency between the Table class and what the real DB looks like. It's just a question of which one is less important.

Another way is to not set a `DEFAULT` at all and still update it manually in the migration class. This will keep the `method` column still as `null` while it's set to be `NOT NULL`. Even though this works on my local MariaDB instance, I'm not sure if that'll work with all versions of MySQL that our users might use.

Let me know what you think.

Reply:
Thanks for looking into this. I agree I went back and forth quite a few times on possible solutions, but I didn't really find any that fit nicely. I think relying on it becoming NULL when it's set to NOT NULL would be risky (although I haven't tested that much).

I'd say to just run with what you have and add a small note to the table definition that the default will be dropped.